### ISSUE TITLE
'show-for-small-only' and 'hide-for-small-only' elements both displaying at a certain width

### ISSUE THREAD

**Issue by hofuchi** [zurb/foundation-sites#10695]
only happens for 1px with font-size and minimum font-size set to larger than default

**Comment by rafibomb** [comment]
Can you provide more details on what screen width you are seeing this? Can you link a codepen demo so it can be replicated?

**Comment by hofuchi** [comment]
![untitled](https://user-images.githubusercontent.com/9373942/32677862-bd2f5142-c625-11e7-84c1-694e85460e0c.png)

```
@media screen and (max-width: 39.9375em) {
  .hide-for-small-only {
    display: none !important; } }

@media screen and (max-width: 0em), screen and (min-width: 40em) {
  .show-for-small-only {
    display: none !important; } }
```

Sorry for the late reply

The picture above shows why they both show up at **large font-sizes only**
At exactly 959px screen width with a 24px font-zize, there's a gap where none of the above media queries are active, I'll see about creating a codepen but it might be hard because it's dependent on screen width and font-size

**Comment by colin-marshall** [comment]
@hofuchi would appreciate a codepen if you have a minute to reproduce it for us. Thanks!

**Comment by hofuchi** [comment]
@colin-marshall 
Here you go:
https://codepen.io/theGrahamScanlon/pen/PRqbjK?css-preprocessor=sass

Since I couldn't get `$global-font-size` working, go to your browser settings and set the font-size to 16px and the minimum-font-size to 16px.

In the codepen, resize the width to between 639px and 640px and both elements show up.

![screen shot 2018-03-11 at 10 03 47](https://user-images.githubusercontent.com/9373942/37254952-963092c0-2513-11e8-8898-0905fd358f13.png)

**Comment by DanielRuf** [comment]
> set the font-size to 16px and the minimum-font-size to 16px

This is the default in most browsers.

I guess you mean https://codepen.io/DanielRuf/pen/eMNgyz?css-preprocessor=sass

**Comment by DanielRuf** [comment]
> between 639px and 640px and both elements show up.

Not sure but I can not do this in Chrome.

**Comment by DanielRuf** [comment]
https://github.com/zurb/foundation-sites/blob/4e21634356da296edd2f1c3e65d91140f991008b/dist/css/foundation.css#L5360
https://github.com/zurb/foundation-sites/blob/4e21634356da296edd2f1c3e65d91140f991008b/dist/css/foundation.css#L5364

**Comment by DanielRuf** [comment]
@SassNinja can we get better / more precise em values for the breakpoints than this 0.0625 difference?

**Comment by hofuchi** [comment]
> I guess you mean https://codepen.io/DanielRuf/pen/eMNgyz?css-preprocessor=sass

I tried, foundation still initializes its base font-size that the media queries use as the browser default, so setting the html, body font-size on codepen won't help reproduce the issue

> Not sure but I can not do this in Chrome.
I was using firefox, but just checked and I can reproduce this in chrome and opera, not sure why you can't

**Comment by DanielRuf** [comment]
> setting the html, body font-size on codepen won't help reproduce the issue

It will. The 1.5 px difference can be seen [here](https://github.com/zurb/foundation-sites/issues/10695#issuecomment-372125107) and [here](https://github.com/zurb/foundation-sites/issues/10695#issuecomment-372124946)

**Comment by DanielRuf** [comment]
> I was using firefox, but just checked and I can reproduce this in chrome and opera, not sure why you can't

I mean that I can not set a viewport with a fraction size.

@ncoden don't we already have an issue about the accuracy of the generated media queries?

**Comment by colin-marshall** [comment]
I can reproduce in FireFox but only if I manually set the width of the CodePen iframe to 639.6px. I don't see a way to resize the viewport window to a non-integer, but I guess there could be situations where you could have an iframe in a page with a decimal width.

**Comment by DanielRuf** [comment]
> Possibly related to #10779

Definitely, it is an accuracy related issue.

**Comment by ncoden** [comment]
It should be fixed by #10779.

**Comment by DanielRuf** [comment]
> It should be fixed by #10779.

Thats the issue, did you want to link a specific PR @ncoden?

**Comment by DanielRuf** [comment]
Did you mean https://github.com/zurb/foundation-sites/pull/10978?
